app/controllers/users_controller.rb

```diff
@@ -8,23 +8,49 @@ def new
     end
   end
 
+  def confirm_email
+    user = User.find_by_confirm_token(params[:id])
+    if user
+      user.email_activate
+      session[:user_id] = user.id
+      redirect_to_role_portal(user)
+    end
+  end
+
   def create
     unless @user = existing_user(params[:user][:email])
       @user = User.new(user_params)
       @user.role = "parent"
     else
+      flash[:alert] = "This email is already in use, please use a different email"
       redirect_to signup_path
     end
 
     if @user.save
-      puts "#{@user.role} SAVED"
-      session[:user_id] = @user.id
-      redirect_to new_parent_parent_path
+      UserMailer.registration_confirmation(@user).deliver
+      flash[:notice] = "Thank you for creating an account. A confirmation email has been sent to the email you provided. Please click on the link to verify your account."
+      redirect_to login_path
     else
+      flash[:alert] = "Oops! Something went wrong, please try again."
       redirect_to signup_path
     end
   end
 
+  def redirect_to_role_portal(user)
+    if user.role == "parent"
+      flash[:notice] = "Thank you for confirming your email! Please fill in the information below!"
+      redirect_to parent_profile_path #LINK TO ADDRESS FORM
+    elsif user.role == "counselor"
+      flash[:notice] = "Thank you for confirming your email. Please set-up your account to continue"
+      redirect_to counselor_settings_path
+    elsif user.role == "director"
+      flash[:notice] = "Thank you for confirming your email! You can now enjoy using CampZone!"
+      redirect_to director_dashboard_index_path
```

Comment:
I would remove these flash messages and create a new method that assigns the flash based on context. 

So it'd be more like this:

```ruby
def create
  ...
  display_flash(@user)
  redirect_to_role_portal(@user)
end

def display_flash(user)
  ...
end

def redirect_to_role_portal(user)
  ...
end
```